thrust/thrust/system/cuda/detail/temporary_buffer.h

```diff
@@ -26,4 +26,107 @@
 #  pragma system_header
 #endif // no system header
 
-// this system has no special temporary buffer functions
+#include <thrust/detail/type_traits/pointer_traits.h>
+#include <thrust/system/cuda/detail/par.h>
+#include <thrust/system/detail/bad_alloc.h>
+
+THRUST_NAMESPACE_BEGIN
+namespace cuda_cub
+{
+
+// If par_nosync does not have a user provided allocator attached, these
+// overloads should be selected.
+
+template <typename T>
+_CCCL_HOST pair<T*, ::cuda::std::ptrdiff_t> get_temporary_buffer(par_nosync_t&, ::cuda::std::ptrdiff_t n)
+{
+  void* ptr;
+  cudaError_t status = cudaMallocAsync(&ptr, sizeof(T) * n, nullptr);
```

Comment:
I believe we should fetch the stream that is provided to `par_nosync` via `on`

Reply:
> I believe we should fetch the stream that is provided to `par_nosync` via `on`

We do. See the overloads below for `execute_with_stream_nosync`, which is the derived execution policy type returned when you do `par_nosync.on(stream)`. This particular overload is only found if no stream has been attached to `par_nosync`.